src/include/migraphx/op/broadcast.hpp

```diff
@@ -46,25 +46,21 @@ struct broadcast
             MIGRAPHX_THROW("BROADCAST : axis is out of range");
         }
 
-        if(std::all_of(
-               broadcast_lens.cbegin(), broadcast_lens.cend(), [&](auto x) { return x == 1; }))
+        if(broadcast_lens.size() - axis < input.lens().size())
         {
-            return {t, broadcast_lens, std::move(bcast_strides)};
+            MIGRAPHX_THROW("BROADCAST: when broadcasting success sizes must match");
```

Comment:
I think the error message is incorrect here

Reply:
fixed